sui/token_bridge/sources/governance/register_chain.move

```diff
@@ -3,269 +3,238 @@ module token_bridge::register_chain {
-        assert!(action == 0x01, E_INVALID_ACTION);
-
-        // TODO(csongor): should we also accept a VAA directly?
-        // Why would a registration VAA target a specific chain?
-        let target_chain = bytes::deserialize_u16_be(&mut cur);
-        assert!(target_chain == 0, E_INVALID_TARGET);
+    public entry fun submit_vaa(
+        token_bridge_state: &mut State,
+        wormhole_state: &mut WormholeState,
+        vaa_buf: vector<u8>,
+        ctx: &TxContext
+    ) {
+        let msg =
+            governance_message::parse_and_verify_vaa(
+                wormhole_state,
+                vaa_buf,
+                ctx
+            );
 
-        let emitter_chain_id = bytes::deserialize_u16_be(&mut cur);
+        // Protect against replaying the VAA.
+        state::consume_vaa_hash(
+            token_bridge_state,
+            governance_message::vaa_hash(&msg)
+        );
 
-        let emitter_address = external_address::deserialize(&mut cur);
+        assert!(
+            governance_message::module_name(&msg) == state::governance_module(),
+            E_INVALID_MODULE
+        );
+        assert!(
+            governance_message::action(&msg) == ACTION_REGISTER_CHAIN,
+            E_INVALID_ACTION
+        );
+        assert!(
+            governance_message::is_global_action(&msg),
+            E_INVALID_TARGET
+        );
 
+        let cur = cursor::new(governance_message::take_payload(msg));
+        let emitter_chain = bytes::take_u16_be(&mut cur);
+        let emitter_address = external_address::take_bytes(&mut cur);
         cursor::destroy_empty(cur);
 
-        RegisterChain { emitter_chain_id, emitter_address }
-    }
-
-    public entry fun submit_vaa(
-        wormhole_state: &mut WormholeState,
-        bridge_state: &mut State,
-        vaa: vector<u8>,
-        ctx: &mut TxContext
-    ) {
-        let vaa = corevaa::parse_and_verify(wormhole_state, vaa, ctx);
-        corevaa::assert_governance(wormhole_state, &vaa);
-        token_bridge_vaa::replay_protect(bridge_state, &vaa);
-        let RegisterChain {
-            emitter_chain_id,
-            emitter_address
-        } = parse_payload(corevaa::destroy(vaa));
-        bridge_state::register_emitter(
-            bridge_state,
-            emitter_chain_id,
+        state::register_emitter(
+            token_bridge_state,
+            emitter_chain,
             emitter_address
         );
     }
 
-    public fun get_emitter_chain_id(a: &RegisterChain): u16 {
-        a.emitter_chain_id
+    #[test_only]
+    public fun emitter_chain(self: &RegisterChain): u16 {
+        self.emitter_chain
     }
 
-    public fun get_emitter_address(a: &RegisterChain): ExternalAddress {
-        a.emitter_address
+    #[test_only]
+    public fun emitter_address(self: &RegisterChain): ExternalAddress {
+        self.emitter_address
     }
 }
 
 #[test_only]
 module token_bridge::register_chain_test {
-    use sui::test_scenario::{
```

Comment:
Do we want to add back the register chain tests in this PR, or handle them in `sui/token-bridge-cleanup`?

Why were they deleted?

Reply:
We'll handle them in the token bridge clean up PR.